olive/engine/engine.py

```diff
@@ -187,6 +197,44 @@ def setup_accelerators(self, execution_providers):
                 "Please consider installing an onnxruntime build that contains the relevant execution providers. "
             )
 
+    def prepare_passes_from_auto_optimizer(
+        self,
+        input_model_config: ModelConfig,
+        evaluator_config: "OliveEvaluatorConfig",
+        accelerator_spec: AcceleratorSpec,
+        auto_optimizer_config: Optional[AutoOptimizerConfig] = None,
+        data_configs: Optional[Dict[str, DataConfig]] = None,
+        output_dir: str = None,
+    ):
+        if not self.enable_auto_optimizer:
+            return
+
+        logger.info("No passes registered, run auto optimizer.")
+        self.cleanup_passes()
+        auto_optimizer = AutoOptimizer(
+            input_model_config,
+            evaluator_config,
+            accelerator_spec,
+            auto_optimizer_config,
+            data_configs=data_configs,
+        )
+        pass_config, pass_flows = auto_optimizer.suggest()
+
+        if pass_config:
+            for pass_name, config in pass_config.items():
+                p = Pass.registry[config["type"].lower()]
+                self.register(p, config["config"], name=pass_name)
+            self.set_pass_flows(pass_flows)
+            # save suggested pass config/pass flows to output_dir/auto_optimizer/{accelerator_spec}.json
```

Comment:
when these files are used? should we use return values so that caller will use them?

Reply:
Removed in the new commit.